sdks/v4-sdk/src/utils/v4Planner.ts

```diff
@@ -117,23 +123,41 @@ export class V4Planner {
   }
 
   addTrade(trade: Trade<Currency, Currency, TradeType>, slippageTolerance?: Percent): void {
-    const actionType = trade.tradeType === TradeType.EXACT_INPUT ? Actions.SWAP_EXACT_IN : Actions.SWAP_EXACT_OUT
+    const exactOutput = trade.tradeType === TradeType.EXACT_OUTPUT
+
+    if (exactOutput) invariant(!!slippageTolerance, 'ExactOut requires slippageTolerance')
+    invariant(trade.swaps.length === 1, 'Only accepts Trades with 1 swap (must break swaps into individual trades)')
+
+    const actionType = exactOutput ? Actions.SWAP_EXACT_OUT : Actions.SWAP_EXACT_IN
 
     const currencyIn = currencyAddress(trade.inputAmount.currency)
     const currencyOut = currencyAddress(trade.outputAmount.currency)
 
-    for (let swap of trade.swaps) {
-      this.addAction(actionType, [
-        {
-          currencyIn,
-          path: encodeRouteToPath(swap.route),
-          amountIn: swap.inputAmount.quotient.toString(),
-          amountOutMinimum: slippageTolerance ? trade.minimumAmountOut(slippageTolerance).quotient.toString() : 0,
-        },
-      ])
-    }
-
-    this.addAction(Actions.SETTLE_TAKE_PAIR, [currencyIn, currencyOut])
+    this.addAction(actionType, [
+      exactOutput
+        ? {
+            currencyOut,
+            path: encodeRouteToPath(trade.route, exactOutput),
+            amountInMaximum: trade.maximumAmountIn(slippageTolerance ?? new Percent(0)).quotient.toString(),
```

Comment:
it looks like on 128 we check that slippageTolerance is required? so can we just pass it in directly here?

Reply:
typescript gets mad

Reply:
hahah ok got it